experimental/uefi/runtime/src/lib.rs

```diff
@@ -34,6 +34,49 @@ pub trait Channel {
     fn recv(&mut self, data: &mut [u8]) -> anyhow::Result<()>;
 }
 
+pub type FrameLength = u16;
+// The frame length is a u16, which is two bytes encoded.
+pub const FRAME_LENGTH_ENCODED_SIZE: usize = 2;
+
+pub struct Framed<T> {
+    inner: T,
+}
```

Comment:
AFAICT it makes no sense to construct a Framed with anything other than an inner T that implements Channel, so perhaps the trait bound should be here too (nothing would change in the impl, but it would be clearer what the expectation is).